agent/agent.go

```diff
@@ -1253,6 +1278,133 @@ func (a *agent) startReportingConnectionStats(ctx context.Context) {
 	}
 }
 
+var prioritizedProcs = []string{"coder"}
+
+func (a *agent) manageProcessPriorityLoop(ctx context.Context) {
+	if val := a.envVars[EnvProcMemNice]; val == "" || runtime.GOOS != "linux" {
+		a.logger.Info(ctx, "process priority not enabled, agent will not manage process niceness/oom_score_adj ",
+			slog.F("env_var", EnvProcMemNice),
+			slog.F("value", val),
+			slog.F("goos", runtime.GOOS),
+		)
+		return
+	}
+
+	manage := func() {
+		procs, err := a.manageProcessPriority(ctx)
+		if err != nil {
+			a.logger.Error(ctx, "manage process priority",
+				slog.F("dir", agentproc.DefaultProcDir),
+				slog.Error(err),
+			)
+		}
+		if a.modifiedProcs != nil {
+			a.modifiedProcs <- procs
+		}
+	}
+
+	// Do once before falling into loop.
+	manage()
```

Comment:
Instead of creating a closure you could add the logic into the begginning of the `for` loop and then make the select statement block empty. E.g.:

```go
	for {
        procs, err := a.manageProcessPriority(ctx)
		if err != nil {
			a.logger.Error(ctx, "manage process priority",
				slog.F("dir", agentproc.DefaultProcDir),
				slog.Error(err),
			)
		}
		if a.modifiedProcs != nil {
			a.modifiedProcs <- procs
		}

		select {
		case <-a.processManagementTick:
            continue
		case <-ctx.Done():
			return
		}
	}
```